test/parallel_api/iterator/transform_output_iterator.pass.cpp

```diff
@@ -0,0 +1,153 @@
+// -*- C++ -*-
+//===-- transform_output_iterator.pass.cpp ---------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "support/utils.h"
+
+#include <tuple>
+
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+DEFINE_TEST(test_copy)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_copy)
+
+    template <typename ExecutionPolicy, typename Iterator1, typename Iterator2, typename Size, typename TExpectedValue>
+    void operator()(ExecutionPolicy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Size n, TExpectedValue expected_value)
+    {
+        TestDataTransfer<UDTKind::eVals, Size> host_vals(*this, n);
+
+        ::std::copy(::std::forward<ExecutionPolicy>(exec), first1, last1, first2);
+
+        host_vals.retrieve_data();
+        auto res_begin = host_vals.get();
+        for(int i = 0; i != n; ++i) {
+            EXPECT_EQ(expected_value, *res_begin, "Wrong result from copy with transform_output_iterator");
+            ++res_begin;
+        }
+    }
+}; // struct test_copy
+
+void test_simple_copy(size_t buffer_size)
+{
+    // 1. create buffers
+    using TestBaseData = test_base_data_buffer<int>;
+    TestBaseData test_base_data({ { buffer_size, 0 },
+                                  { buffer_size, 0 } });
+
+    // 2. create iterators over source buffer 
+    auto sycl_source_begin = test_base_data.get_start_from(UDTKind::eKeys);
+
+    // 3. run algorithms
+    auto transformation = [](int item) { return item + 1; };
+
+    int identity = 0;
+    auto& sycl_src_buf = test_base_data.get_buffer(UDTKind::eKeys);
+    auto host_source_begin = sycl_src_buf.get_access<sycl::access::mode::write>().get_pointer();
```

Comment:
This line contains error: you are using temporary accessor, from which you return some pointer and after that accessor will be destroyed. So usage of get_pointer like this is incorrect.
Please save accessor into separate local variable and then use call of get_pointer from this local variable.

Reply:
All of these have been fixed. I have also fixed the same instances in transform_iterator.pass.cpp